src/ui/ui_nbgl.c

```diff
@@ -319,51 +338,61 @@ static void review_msg_certif_choice(bool confirm) {
     }
 }
 
-static void single_action_msg_certif_review(void) 
+static void single_action_msg_cert_review(void) 
 {
     // Setup data to display
-    msg_certif_pairs[0].item = "Message hash";
-    msg_certif_pairs[0].value = (const char *)fullAddress;
+    msg_cert_pairs[0].value = (const char *)fullAddress;
+    if(sign_type == SIGN_MESSAGE)
+    {
+        msg_cert_pairs[0].item = "Message hash";
+    }
+    else
+    {
+        msg_cert_pairs[0].item = "Certificate hash";
+    }
 
     // Setup list
-    msg_certif_pair_list.nbMaxLinesForValue = 0;
-    msg_certif_pair_list.nbPairs = MSG_CERTIF_MAX_TAG_VALUE_PAIRS_DISPLAYED;
-    msg_certif_pair_list.pairs = msg_certif_pairs;
+    msg_cert_pair_list.nbMaxLinesForValue = 0;
+    msg_cert_pair_list.nbPairs = MSG_CERT_MAX_TAG_VALUE_PAIRS_DISPLAYED;
+    msg_cert_pair_list.pairs = msg_cert_pairs;
 
     // Info long press
-    msg_certif_info_long_press.icon = &C_Message_64px;
-    if(transaction_type == MSG_TRANSACTION)
+    msg_cert_info_long_press.icon = &C_Message_64px;
+    msg_cert_info_long_press.longPressText = "Hold to sign";
+    if(sign_type == SIGN_MESSAGE)
     {
-        msg_certif_info_long_press.text = "Sign message?";
+        msg_cert_info_long_press.text = "Sign message?";
+        nbgl_useCaseStaticReview(&msg_cert_pair_list, &msg_cert_info_long_press, "Reject message", review_msg_cert_choice);
```

Comment:
"reject message" should be replaced by "reject transaction".
The bottom button should be always "reject transaction" (generic wording)

Ok for the wording change of `msg_cert_info_long_press.text`